integration/deno-compiler-test.ts

```diff
@@ -0,0 +1,231 @@
+
+const importPattern = (importSpecifier: string) =>
+  new RegExp(
+    String.raw`import\s*{.*}\s*from\s*"` + importSpecifier + String.raw`"`
+  );
+
+const findCodeFiles = async (directory: string) =>
+  glob.sync("**/*.@(js|jsx|ts|tsx)", {
+    cwd: directory,
+    absolute: true,
+  });
+const searchFiles = async (pattern: string | RegExp, files: string[]) => {
+  let result = shell.grep("-l", pattern, files);
+  return result.stdout
+    .trim()
+    .split("\n")
+    .filter((line) => line.length > 0);
+};
+
+test.beforeAll(async () => {
+  projectDir = await createFixtureProject({
+    template: "deno-template",
+    files: {
+      "package.json": json({
+        private: true,
+        sideEffects: false,
+        dependencies: {
+          "@remix-run/deno": "0.0.0-local-version",
+          "@remix-run/react": "0.0.0-local-version",
+          react: "0.0.0-local-version",
+          "react-dom": "0.0.0-local-version",
+          component: "0.0.0-local-version",
+          "deno-pkg": "0.0.0-local-version",
+        },
+        devDependencies: {
+          "@remix-run/dev": "0.0.0-local-version",
+        },
+      }),
+      "app/routes/index.jsx": js`
+        import fake from "deno-pkg";
+        import { urlComponent } from "https://deno.land/x/component.ts";
+        import { urlUtil } from "https://deno.land/x/util.ts";
+        import { urlServerOnly } from "https://deno.land/x/server-only.ts";
+
+        import { npmComponent } from "npm-component";
+        import { npmUtil } from "npm-util";
+        import { npmServerOnly } from "npm-server-only";
+
+        import { useLoaderData } from "@remix-run/react";
+
+        export const loader = () => {
+          return json({
+            a: urlUtil(),
+            b: urlServerOnly(),
+            c: npmUtil(),
+            d: npmServerOnly(),
+          });
+        }
+
+        export default function Index() {
+          const data = useLoaderData();
+          return (
+            <ul>
+              <li>{fake}</li>
+
+              <li>{urlComponent}</li>
+              <li>{urlUtil()}</li>
+              <li>{data.a}</li>
+              <li>{data.b}</li>
+
+              <li>{npmComponent}</li>
+              <li>{npmUtil()}</li>
+              <li>{data.c}</li>
+              <li>{data.d}</li>
+            </ul>
+          )
+        }
+      `,
+      "node_modules/npm-component/package.json": json({
+        name: "component",
+        version: "1.0.0",
+        sideEffects: false,
+      }),
+      "node_modules/npm-component/index.js": js`
+        module.exports = { npmComponent: () => "NPM_COMPONENT" };
+      `,
+      "node_modules/npm-util/package.json": json({
+        name: "util",
+        version: "1.0.0",
+        sideEffects: false,
+      }),
+      "node_modules/npm-util/index.js": js`
+        module.exports = { npmUtil: () => "NPM_UTIL" };
+      `,
+      "node_modules/npm-server-only/package.json": json({
+        name: "server-only",
```

Comment:
Shouldn't this be `npm-server-only` to be correct? 🤔

```suggestion
        name: "npm-server-only",
```